src/Microsoft.SqlTools.ServiceLayer/Management/Common/DataContainer.cs

```diff
@@ -1204,18 +1205,20 @@ private static XmlDocument CreateDataContainerDocument(ConnectionInfo connInfo,
             }
             else
             {
+                var databaseNode = new XElement("database");
+                databaseNode.Value = connInfo.ConnectionDetails.DatabaseName;
                 xml =
                 string.Format(@"<?xml version=""1.0""?>
                 <formdescription><params>
                 <servername>{0}</servername>
                 <connectionmoniker>{0} (SQLServer, user = {1})</connectionmoniker>
                 <servertype>sql</servertype>
                 <urn>Server[@Name='{0}']</urn>
-                <database>{2}</database>                
+                {2}                
                 </params></formdescription> ",
                 connInfo.ConnectionDetails.ServerName.ToUpper(CultureInfo.InvariantCulture),
                 connInfo.ConnectionDetails.UserName,
-                connInfo.ConnectionDetails.DatabaseName);
+                databaseNode);
```

Comment:
Shouldn't we be doing the same for the other nodes that have custom values? While I believe that currently we don't allow special characters like > in them that isn't something we can guarantee forever so seems reasonable to be defensive here.

We should also be escaping single quotes in the Server name passed to the urn node (again, just to be safe)